src/ts/contentScripts/live-preview/index.tsx

```diff
@@ -0,0 +1,279 @@
+import {Feature, Settings} from '../../utils/settings'
+import {Navigation} from '../../roam/navigation'
+import {browser} from 'webextension-polyfill-ts'
+import {createPopper, Instance} from '@popperjs/core'
+
+export const config: Feature = {
+    id: 'live_preview',
+    name: 'Live Preview',
+    warning: 'Experimental feature; Large databases might see performance issues.',
+    enabledByDefault: false,
+}
+const checkSettingsAndSetupIframe = () => {
+    Settings.isActive('live_preview').then(active => {
+        setupIframe(active)
+    })
+}
+checkSettingsAndSetupIframe()
+
+browser.runtime.onMessage.addListener(async message => {
+    if (message === 'settings-updated') {
+        checkSettingsAndSetupIframe()
+    }
+})
+
+let iframeInstance: PreviewIframe | null = null
+
+const setupIframe = (active: boolean) => {
```

Comment:
this name is misleading (i.e. it does not reflect that it'll destroy the preview if called). `togglePreview` maybe?